src/compile/data/assemble.ts

```diff
@@ -156,6 +156,19 @@ function makeWalkTree(data: VgData[]) {
    * Recursively walk down the tree.
    */
   function walkTree(node: DataFlowNode, dataSource: VgData) {
+    if (node instanceof ParseNode) {
+      if (node.parent instanceof SourceNode && !node.parent.parent)  {
+        // If its parent is a root source node, use normal format parse
+        dataSource.format = {
```

Comment:
Yeah, that's the bug I was talking about yesterday. Which is exactly why I added the parse expression stuff. Basically, when there si a fork, you have to use parse as expressions.